src/components/UserSelector.tsx

```diff
@@ -1,26 +1,29 @@
-import React, { useContext, useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
 import { User } from '../types/User';
+import * as usersActions from '../features/users';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { authorSlice } from '../features/author';
 
 type Props = {
   value: User | null;
-  onChange: (user: User) => void;
 };
 
 export const UserSelector: React.FC<Props> = ({
   // `value` and `onChange` are traditional names for the form field
   // `selectedUser` represents what actually stored here
   value: selectedUser,
-  onChange,
 }) => {
   // `users` are loaded from the API, so for the performance reasons
   // we load them once in the `UsersContext` when the `App` is opened
   // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
+  const users = useAppSelector(state => state.users);
   const [expanded, setExpanded] = useState(false);
+  const dispatch = useAppDispatch();
 
   useEffect(() => {
+    dispatch(usersActions.loadUsers());
+
```

Comment:
The `useEffect` hook is used to handle side effects in your component. In this case, it's used to dispatch the `loadUsers` action. However, it's generally recommended to dispatch actions that load data from the API in the `useEffect` hook of the parent component or in the component where the data is actually needed. This way, the data is loaded when the component is mounted and you avoid unnecessary re-renders.